runtime/syntax/c.yaml

```diff
@@ -5,18 +5,20 @@ detect:
 
 rules:
     - identifier: "\\b[A-Z_][0-9A-Z_]+\\b"
-    - type: "\\b(float|double|bool|char|int|short|long|enum|void|struct|union|typedef|(un)?signed|inline)\\b"
-    - type: "\\b((s?size)|((u_?)?int(8|16|32|64|ptr)))_t\\b"
+    - type: "\\b(_Atomic|_BitInt|float|double|_Decimal32|_Decimal64|_Decimal128|_Complex|complex|_Imaginary|imaginary|_Bool|bool|char|int|short|long|enum|void|struct|union|typedef|typeof|typeof_unqual|(un)?signed|inline|_Noreturn)\\b"
```

Comment:
Just noting that `complex`, `imaginary`, and `offsetof` (below) aren't *technically keywords, even in C23, but they effectively function as such in practice.

* https://en.cppreference.com/w/c/numeric/complex/imaginary
* https://en.cppreference.com/w/c/numeric/complex/complex
* https://en.cppreference.com/w/cpp/types/offsetof

Reply:
The last point should refer to the [c-version](https://en.cppreference.com/w/c/types/offsetof) of `offsetof` or am I wrong? :wink:

Reply:
Right, though they're basically the same in all implementations anyhow.

Reply:
> What do you think about that?

I'm planning to contribute more syntax updates after this PR, so yes, I'll very likely get to the C++ syntax as well.